frontend/src/models/tags.ts

```diff
@@ -1,96 +1,212 @@
 import { createModel } from '@rematch/core'
-import { DEFAULT_TARGET } from '../shared/constants'
-import { graphQLRequest, apiError } from '../services/graphQL'
+import { AxiosResponse } from 'axios'
+import { eachSelectedDevice } from '../helpers/selectedHelper'
+import {
+  graphQLSetTag,
+  graphQLAddTag,
+  graphQLRemoveTag,
+  graphQLDeleteTag,
+  graphQLRenameTag,
+} from '../services/graphQLMutation'
+import { findTagIndex, mergeTags } from '../helpers/utilHelper'
+import { graphQLBasicRequest } from '../services/graphQL'
+import { getNextLabel } from './labels'
 import { RootModel } from './rootModel'
 
-type ITagState = ILookup<ITag[]> & {
+type ITagState = {
   all: ITag[]
+  legacy: ITag[]
+  adding?: boolean
+  removing?: boolean
+  deleting?: string
+  updating?: string
 }
 
 const defaultState: ITagState = {
-  all: [
-    {
-      id: 1001,
-      name: 'Gray',
-      label: 1,
+  all: [],
+  legacy: [],
+}
+
+export default createModel<RootModel>()({
+  state: { ...defaultState },
+  effects: dispatch => ({
+    async fetch(_, globalState) {
+      const { legacy } = globalState.tags
+      const result = await graphQLBasicRequest(
+        ` query {
+            login {
+              tags {
+                name
+                color
+                created
+              }
+            }
+          }`
+      )
+      if (result === 'ERROR') return
+      const parsed = await dispatch.tags.parse(result)
+      const all = mergeTags(legacy, parsed)
+      dispatch.tags.setOrdered({ all })
     },
-    {
-      id: 1002,
-      name: 'Red',
-      label: 2,
+
+    async parse(result: AxiosResponse<any> | undefined, globalState) {
+      const all = result?.data?.data?.login?.tags
+      const parsed = all.map(t => ({
+        name: t.name,
+        color: t.color,
+        created: new Date(t.created),
+      }))
+      return parsed
     },
-    {
-      id: 1003,
-      name: 'Orange',
-      label: 3,
+    // async migrate(_, globalState) {
```

Comment:
Did you mean to leave this in here?

Reply:
Yea just so it's saved in git... then will remove - thanks!